src/shared/copy.c

```diff
@@ -844,8 +847,12 @@ static int copy_fs_verity(int fdf, int *fdt) {
                 .salt_ptr = (uintptr_t) &desc.salt,
         };
 
-        if (ioctl(*fdt, FS_IOC_ENABLE_VERITY, &enable_arg) < 0)
-                return log_error_errno(errno, "Failed to set fs-verity metadata: %m");
+        if (ioctl(*fdt, FS_IOC_ENABLE_VERITY, &enable_arg) < 0) {
+                log_error_errno(errno, "Failed to set fs-verity metadata: %m");
+                if (errno == ENOTTY || errno == EOPNOTSUPP)
+                        errno = ESOCKTNOSUPPORT;
```

Comment:
please don't touch errno unnecessarily. we conisder it mostly read-only.  i.e. do `return -ESOCKTNOSUPPORT;` here.